### DISCUSSION TITLE
One chain cases

### DISCUSSION THREAD

**Discussion by gcroci2** [DeepRank/deeprank2#604] [Q&A] [answered]
A user asked whether the software is usable with PDB files containing proteins with only one chain (i.e., only single proteins and not complexes). I think that in principle this can be handled by deeprank2, but then the question is which query class could be used. I think that none of the ones we have at the moment are meant for such cases (i.e., featurize individual proteins for training a predictor on them). The `ProteinProteinInterfaceQuery` will raise a ValueError if less than 2 chains are passed, and the `SingleResidueVariantQuery` requires a `variant_residue_number` when creating the instance (please correct me if I am wrong @DaniBodor @cbaakman).

Do you think it would make sense to add a class for such cases? Could they be of enough interest to the community? @LilySnow

**Comment by DaniBodor** [comment]
It depends what the research question is that this person wants to address?

Our `Query` classes are based on the research question. `ProteinProteinInterfaceQuery` is used for the question "will these two chains bind?", so inherently requires (at least) two chains. `SingleResidueVariantQuery` is used for the question "will this variant have a functional consequence", so inherently requires a variation.

I guess if one could ask about intra-chain bonding, is that the question here?. I'm sure that there could be smart way to deal with this by the code base, but as a workaround for now I think it could make sense to assign the different chain ids to the two side of the binding interface. I can't immediately think of a reason why this would be problematic, but I might be proven wrong...

**Reply by gcroci2** [reply]
> ProteinProteinInterfaceQuery is used for the question "will these two chains bind?", 

Not only actually, the question can be whatever classification (both binary and multi-classification) or regression question.
Regarding single proteins cases, similarly: is this protein involved in some process X yes/no(classification)? Or what is the physical value of property X for this protein (regression)? In such cases, relying on the current classes would be not ideal, even if an hacky workaround would be relatively possible. However my question is more to understand how much interest in the field there would be for such cases, so if practically speaking the majority of reasearch that makes use of supervised learning in the field is interested in biomolecular complexes interactions rather than single biomolecules predictions (either for classification or regression).

**Reply by DaniBodor** [reply, marked as answer]
You're right. I guess I was sidetracked a bit by the name of the query class. 
It wouldn't be difficult to change the query class to also allow single chains (and it should then also be renamed) or add an additional class for that.
On the question of how much interest there is for it, I cannot gauge. My guess would be that there is enough interest to warrant the relatively simple change/addition, but that is just a guess.

**Reply by gcroci2** [reply]
I agree that it should be relatively simple. I opened an issue for it (#605), it may fit in the SS budget - and scope.
